components/map/basic/src/index.js

```diff
@@ -0,0 +1,227 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+
+class MapBasic extends Component {
+  constructor (props) {
+    super(props)
+    this.map = this._createMapObject()
+    this._setCurrentLoadedLibraries()
+    this._setCurrentLoadedStyles()
+    this._initLibraryInjection(props.libraries, props.appId, props.appCode, props.useCIT, props.useHTTPS)
+    this._initStylesInjection(props.styles)
+  }
+
+  _createMapObject () {
+    return {
+      instance: null,
+      behavior: null,
+      ui: null,
+      platform: null,
+      defaultLayers: null
+    }
+  }
+
+  _setCurrentLoadedStyles () {
+    this.currentStylesLoaded = Array.from(document.querySelectorAll('link')).map(style => style.href)
+  }
+
+  _setCurrentLoadedLibraries () {
+    this.currentLibrariesLoaded = Array.from(document.querySelectorAll('script')).map((library) => library.src)
```

Comment:
Always optimize your selectors and be as specific as possible.
Look at the performance of `querySelectorAll` when you could use `getElementsByTagName` instead

https://jsperf.com/queryselectorall-vs-getelementsbytagname